cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -92,7 +92,16 @@ type Etcd struct {
 	DataDir   string
 	ExtraArgs map[string]string
 	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image string
+	Image      string
+	SelfHosted SelfHostedEtcd
```

Comment:
make this a pointer

Reply:
we should really soon make the v1alpha2 API: https://docs.google.com/document/d/1LNaGKzjyLtPnLbRbW4QUfwVEJS3XcyQy_JmWyV-bVbU/edit#heading=h.zd8ht9pthabe

(and then split the Etcd struct in Local, External and SelfHosted)